File: Demos/VB.Net/CustomOpenSaveDialogs/OpenDialogForm.vb
Public Class OpenDialogForm
    Public ReadOnly Property ReportName() As String
        Get
            Return CStr(lbxReports.SelectedItem)
        End Get
    End Property

    Public WriteOnly Property ReportsTable() As DataTable
        Set(ByVal value As DataTable)
            Dim row As DataRow
            For Each row In value.Rows
                lbxReports.Items.Add(row.Item("ReportName"))
            Next
        End Set
    End Property

    Private Sub lbxReports_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles lbxReports.SelectedIndexChanged
        btnOK.Enabled = Not String.IsNullOrEmpty(Me.ReportName)
    End Sub
End Class

File: Demos/C#/CustomOpenSaveDialogs/OpenDialogForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace CustomOpenSaveDialogs
{
  public partial class OpenDialogForm : Form
  {
    public DataTable ReportsTable
    {
      set
      {
        // fill the listbox with names of reports
        foreach (DataRow row in value.Rows)
        {
          lbxReports.Items.Add(row["ReportName"]);
        }
      }
    }

    public string ReportName
    {
      get
      {
        return (string)lbxReports.SelectedItem;
      }
    }
    
    public OpenDialogForm()
    {
      InitializeComponent();
    }

    private void lbxReports_SelectedIndexChanged(object sender, EventArgs e)
    {
      btnOK.Enabled = !String.IsNullOrEmpty(ReportName);
    }
  }
}